filters/openpolicyagent/openpolicyagent.go

```diff
@@ -453,6 +453,9 @@ func (registry *OpenPolicyAgentRegistry) startCustomControlLoopDaemon() {
 			registry.mu.Unlock()
 
 			for _, opa := range instances {
+				if opa != nil && opa.Healthy() { // Skip retriggering plugins of healthy instance
```

Comment:
This would prevent bundle updates for healthy instances, i.e. they would stop fetching. I don’t think we want this…

Reply:
Thanks for catching this. It indeed depend on this loop since trigger is set to manual mode.

There was a data race without this between control loop and parallel start. 
[parallelVSloopRACE.txt](https://github.com/user-attachments/files/22454030/parallelVSloopRACE.txt) will see if `!opa.Started` can be a solution here. 

Still another race condition remained with this, 
[DownloadRACE.txt](https://github.com/user-attachments/files/22454047/DownloadRACE.txt) 
The etag value of the Downloader is 
written as
 
```
func (d *Downloader) SetCache(etag string) {
	d.etag = etag
}
``` 

and read as
```	
d.client = d.client.WithHeader("If-None-Match", d.etag)
```
without synchronization in OPA download.go file.